### PR TITLE
feat: Add support for setting the condition field in Event Bus permissions

### PR DESCRIPTION
**Pull request by cadrake**
## Description
This PR adds a new `condition_org` field to the `permissions` maps that will be used to set the `condition` section of the `aws_cloudwatch_event_permission` resource to allow limiting access to an EventBridge bus.

## Motivation and Context
I wanted to be able to support more narrow permissions on an EventBridge bus akin to the examples shown in the terraform provider docs. Since complete map of `key`, `type` and `value` can't be passed in without making the entire variable an object, but `key` and `type` only have a single value, I opted to make the `value` the only field and set the remaining ones to their defaults when generating.

## Breaking Changes
This change is backwards compatible

## How Has This Been Tested?
I modified the examples/with-permission module to deploy an event bus with access limited to only a single organization and then used the cli to produce an event to the bus.

- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -216,6 +216,18 @@ resource "aws_cloudwatch_event_permission" "this" {
 
   action         = lookup(each.value, "action", null)
   event_bus_name = try(each.value["event_bus_name"], aws_cloudwatch_event_bus.this[0].name, var.bus_name, null)
+
+  dynamic "condition" {
+    for_each = lookup(each.value, "condition_org", null) != null ? [
+      each.value.condition_org
```
Comment (antonbabenko) [inline main.tf:222]:
```suggestion
```

File: main.tf [hunk 2]
```diff
@@ -216,6 +216,18 @@ resource "aws_cloudwatch_event_permission" "this" {
...
+    ] : []
```
Comment (antonbabenko) [inline main.tf:223]:
```suggestion
```

**Review by antonbabenko — COMMENTED** [review]
Looks pretty good. Could you please verify and confirm that my proposed changes work?

**Comment by cadrake** [comment]
I made your changes and applied the example again, everything looks good.

**Comment by antonbabenko** [comment]
This PR is included in version 2.1.0 :tada:

**Pull request merged**